### PR TITLE
Documentation - allow long lines for tables and code blocks

### LINKED ISSUE #2917: Globally allow long lines for code_blocks and tables
# Feature Request

Create markdown lint configuration file.
Disable MD013 check for code blocks and tables.
Remove all MD013 enable/disable comments from doc files.

Based on https://github.com/open-telemetry/opentelemetry-dotnet/pull/2914#pullrequestreview-886931323

### PR DESCRIPTION
**Pull request by Kielek**
Fixes #2917.

## Changes

Introduce md lint config file.
Allow long lines for code-block and tables.
Remove obsolete markups.

### CODE REVIEW

File: .github/workflows/markdownlint.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: markdownlint
 
 on:
   push:
-    branches: [ main, metrics ]
+    branches: [ long-lines-linting ]
```
Comment (cijothomas) [inline .github/workflows/markdownlint.yml:5]: intentional? I guess not..

Comment (Kielek) [reply .github/workflows/markdownlint.yml]: Sure no... I have attached wrong file to the first commit.
It should be in another one, used to execute pipeline on fork.

Fixed afc2d71a4d3091e62cc9a9721edff8e8a3b0c43d

Comment (Kielek) [reply .github/workflows/markdownlint.yml]: It was used to test CI pipeline on my fork. It should be included in the commit which is not part of this PR.

Fixed in afc2d71a4d3091e62cc9a9721edff8e8a3b0c43d

**Review by reyang — APPROVED** [review]
Nice improvements! 👍 LGTM.

**Review by CodeBlanch — APPROVED** [review]
❤️

**Pull request merged**